### PR TITLE
fixing maxWaitQueueSize set by ConnectionPoolSettingsParser

### PR DESCRIPTION
**Pull request by xarv**
At line 39 in ConnectionsPoolSettings.java the maxWaitQueueSize is being set as waitQueueMultiple fixing that

### CODE REVIEW

File: vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/config/ConnectionPoolSettingsParser.java [hunk 1]
```diff
@@ -36,7 +36,8 @@ public ConnectionPoolSettingsParser(ConnectionString connectionString, JsonObjec
       }
       Integer waitQueueMultiple = config.getInteger("waitQueueMultiple");
       if (waitQueueMultiple != null) {
-        settings.maxWaitQueueSize(waitQueueMultiple);
+	Integer waitQueueSize = waitQueueMultiple * DEFAULT_MONGO_DRIVER_WAIT_Q_SIZE;
```
Comment (karianna) [inline vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/config/ConnectionPoolSettingsParser.java]: White space indentation issue on this line

**Comment by karianna** [comment]
I'm not sure if this is a correct fix.  The docs state:

> waitQueueMultiple is a number that the driver multiples the maxPoolSize value to, to provide the maximum number of threads allowed to wait for a connection to become available from the pool.

So this PR changes the queue of threads waiting from size of `x` to be size `x * 100`.  That's a lot of waiting threads...

**Comment by xarv** [comment]
The documentation mentions it but the implementation inadvertently initialises     maxWaitQueueSize as waitQueueMultiple by calling [this](https://github.com/mongodb/mongo-java-driver/blob/master/driver-core/src/main/com/mongodb/connection/ConnectionPoolSettings.java#L105) of the builder object inside the mongo driver. And yes the default wait queue size is 500 rather than 100, let me know if this seems ok i will modify the constant.

File: vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/config/ConnectionPoolSettingsParser.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
 class ConnectionPoolSettingsParser {
 
   private final ConnectionPoolSettings settings;
-
+  private final Integer DEFAULT_MONGO_DRIVER_WAIT_Q_SIZE=100;
```
Comment (karianna) [inline vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/config/ConnectionPoolSettingsParser.java]: OK, so maybe set this to 500 and reference the mongo driver variable in a comment.

**Comment by karianna** [comment]
OK, I think make the change to 500 as the default (with a reference to that private constant in a comment), fix the small white space issue and then I think this is OK.

**Comment by xarv** [comment]
updated.

**Pull request merged**